opengovsg/postmangovsg
# chore: Add Sentry monitoring for Telegram handler

## Problem

Closes #554 

## Solution

**Features**:

- Added Sentry error capturing for `telegram-handler`

## Tests

- Check that Sentry DSN is set to the `postmangovsg-serverless` project and `NODE_ENV` is set to `development`
- Insert a valid credential that is added to the database 
- Run dev server with `npm run dev`
- Send requests using `curl` to trigger the following scenarios:
    1. `Invalid bot token`: Use an invalid token
    2. `[BOT_ID] not recognized`: Use a token that contains a `botId` that is not in the database
    3. `Sender and contact mismatch`: Post the following payload to the webhook endpoint with a valid bot token
```
{
   "message":{
      "from":{
         "id":[YOUR TELEGRAM ID]
      },
      "contact":{
         "phone_number": "[PHONE NUMBER]",
         "first_name": "Hello",
         "last_name": "World",
         "user_id":123456
      }
   }
}
```

## Deploy Notes

**New environment variables**:

- `SENTRY_DSN` : DSN for `postmangovsg-serverless` project
- `SENTRY_LAMBDA_FUNCTION_NAME`: Lambda function name used to tag a Sentry issue for identification

**New dependencies**:

- `@sentry/node` : Sentry SDK
- `source-map-support`: Library to support TS stack traces without explicitly uploading source maps

**TODOs**
- [x] Set Sentry alert rules for `postmangovsg-serverless`
- [ ] Test alert rules after merge
